pkg/server/server.go

```diff
@@ -520,16 +530,24 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			connID := pkt.GetData().ConnectID
 			data := pkt.GetData().Data
 			klog.V(5).InfoS("Received data from connection", "bytes", len(data), "connectionID", connID)
+			if backend == nil {
+				klog.V(2).InfoS("Backend has not been initialized for the connection. Client should send a Dial Request first", "connectionID", connID)
+				s.sendFrontendClose(stream, connID, "backend not initialized")
+				return
+			}
+
 			if firstConnID == 0 {
 				firstConnID = connID
 			} else if firstConnID != connID {
-				klog.V(5).InfoS("Data does not match first connection id", "fistConnectionID", firstConnID, "connectionID", connID)
+				klog.ErrorS(nil, "Data does not match first connection id", "fistConnectionID", firstConnID, "connectionID", connID)
```

Comment:
fist?

Reply:
Do we believe we have seen this? If so I would very much like to see a metric added.

Reply:
Fixed. This could happen if a frontend client was attempting to reuse a tunnel. Agree on the metric, but I'd add it as part of a generic `connection_closed` metric with a `close_reason`. Basically the post-dial equivalent of the `dial_failures` metric: https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/410

Reply:
Not sure even a reuse will cause a new connID. the tunnel sets the connID while the tunnel is being set up. 
The connection ID is set by the agent at https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/agent/client.go#L404.
The means it will only be created in response to a dial request which is created by the client at https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/cec42080fe55c2d21a2dfde35314f5f05bc65b85/konnectivity-client/pkg/client/client.go#L410.
Within the client that is only called from CreateSingleUseGrpcTunnelWithContext (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/cec42080fe55c2d21a2dfde35314f5f05bc65b85/konnectivity-client/pkg/client/client.go#L169).
So reusing a connection on a tunnel should not create a new connection id.